src/sidebar/panel.js

```diff
@@ -334,95 +114,26 @@ closeButton.addEventListener('click', () => {
   noteDiv.classList.toggle('visible');
 });
 
-enableSync.onclick = () => {
-  noteDiv.classList.toggle('visible');
-  browser.runtime.sendMessage({
-    action: 'authenticate',
-    context: getPadStats()
-  });
-};
-
-// gets the user-selected theme from local storage and applies respective CSS
-// file to the document
-function getThemeFromStorage() {
-  const getting = browser.storage.local.get(['theme']);
-  getting.then(function applyTheme(data) {
-    if (data.theme === 'dark') {
-      if (! document.getElementById('dark-styles')) {
-        const darkSS = document.createElement('link');
-        darkSS.id = 'dark-styles';
-        darkSS.type = 'text/css';
-        darkSS.rel = 'stylesheet';
-        darkSS.href = 'styles-dark.css';
-        document.getElementsByTagName('head')[0].appendChild(darkSS);
-      } else
-        return;
-    } else if (data.theme === 'default' || data.theme === undefined) {
-      if (document.getElementById('dark-styles')) {
-        const darkSS = document.getElementById('dark-styles');
-        darkSS.parentElement.removeChild(darkSS);
-      } else
-        return;
-    }
-  });
-}
-document.addEventListener('DOMContentLoaded', getThemeFromStorage);
-
-chrome.runtime.onMessage.addListener(eventData => {
-  switch (eventData.action) {
-    case 'text-change':
-      ignoreNextLoadEvent = true;
-      loadContent();
-      break;
-    case 'theme-changed':
-      getThemeFromStorage();
-  }
-});
-
-// disable drop of links and images into notes
-const qlEditor = document.querySelectorAll('.ql-editor');
-
-document.addEventListener('dragover', () => {
-  qlEditor[0].classList.add('forbid-cursor');
-  browser.runtime.sendMessage({
-    action: 'metrics-drag-n-drop',
-    context: getPadStats()
-  });
-  return true;
-});
-
-document.addEventListener('dragleave', () => {
-  qlEditor[0].classList.remove('forbid-cursor');
-  return true;
-});
-
-document.addEventListener('drop', (e) => {
-  e.preventDefault();
-  qlEditor[0].classList.remove('forbid-cursor');
-  return false;
-});
-
-function getPadStats() {
-  const content = quill.getContents();
-  const text = quill.getText();
+function getPadStats(editor) {
+  // const content = quill.getContents();
+  const text = editor.getData();
   const styles = {
     size: false,
     bold: false,
-
     italic: false,
     strike: false,
     list: false
   };
 
-  content.forEach(node => {
-    if (node.hasOwnProperty('attributes')) {
-      Object.keys(node.attributes).forEach(key => {
-        if (styles.hasOwnProperty(key)) {
-          styles[key] = true;
-        }
-      });
-    }
-  });
+  // content.forEach(node => {
```

Comment:
@szymonkups, @Reinmar Need a bit of help here. We used to look at node attributes in Quill to detect of features are being used. How do we detect if `bold`, `italic`, etc are used? `getData` just returns text. In Quill there is quill.getContents(); vs quill.getData();

Reply:
It'd be something like:

```js
import Range from '@ckeditor/ckeditor5-engine/src/model/range';

const range = Range.createIn( editor.document.getRoot() );

for ( const value of range ) {
    if ( value.item.is( 'text' ) ) {
        console.log( Array.from( value.item.getAttributes() ) );
    } 
}
```

Iterating over a [range](https://ckeditor5.github.io/docs/nightly/ckeditor5/latest/api/module_engine_model_range-Range.html) uses a [tree walker](https://ckeditor5.github.io/docs/nightly/ckeditor5/latest/api/module_engine_model_treewalker-TreeWalker.html) and `value` is an instance of [`TreeWalkerValue`](https://ckeditor5.github.io/docs/nightly/ckeditor5/latest/api/module_engine_model_treewalker-TreeWalkerValue.html).

This will print which text attributes are used. You can also check which elements are used by `console.log( value.item.name )` if `value.item.is( 'element' )` is true.

Of course, this means that you need to build this code together with CKEditor. So you can't use a built version of CKEditor, but build CKEditor from source together with the code which uses it. It's explained here: https://ckeditor5.github.io/docs/nightly/ckeditor5/latest/framework/guides/quick-start.html.

We're now working on [making it simpler to write some basic features without building CKEditor](https://github.com/ckeditor/ckeditor5/issues/667). This means exposing more APIs somewhere in the `editor` properties.

PS. Why do you need to check which features are used in the editor?

Reply:
> PS. Why do you need to check which features are used in the editor?

So we can learn which features are the most popular. User research, etc.. :)

Reply:
@Natim wanna take a look at this? 

need to uncomment the `content.forEach(node => {` function, otherwise metrics works.

Reply:
"uncomment" => make it work. You might need to add a dependency that includes the `Range` module. 

> build CKEditor from source together with the code which uses it

Reply:
@Natim Our build script is here: https://github.com/mozilla/notes/pull/394/files#diff-bef1bc0d0ce8255123c3de896c5a796b , builds and copies with `npm run postinstall`

Reply:
It doesn't seems to work:

```
ERROR in ./src/ckeditor.js
Module not found: Error: Can't resolve '@ckeditor/ckeditor5-editor-classic/src/classiceditor' in '/home/rhubscher/mozilla/notes/ckeditor-build/src'
 @ ./src/ckeditor.js 6:0-85
```

Reply:
Trying to install ``npm install --save @ckeditor/ckeditor5-editor-classic``

Reply:
`npm install --save @ckeditor/ckeditor5-build-classic` nailed it.